src/main/java/com/doit/api/user/UserService.java

```diff
@@ -10,56 +15,66 @@
+
+    public final static String USER_NOT_FOUND_MSG = "user with email %s not found";
+
 
-    // TODO remove this
     @PostConstruct
     // add some users if db is empty
     void init() {
-        findByEmail("user@abc.com")
-        .map(user -> user)
-        .orElseGet(() -> createUser(
-                User.builder()
-                    .email("user@abc.com")
-                    .isEnabled(true)
-                    .isAccountNonExpired(true)
-                    .isAccountNonLocked(true)
-                    .isCredentialsNonExpired(true)
-                    .password("user")
-                    .build()));
-        findByEmail("admin@abc.com")
-        .map(user -> user)
-        .orElseGet(() -> createUser(User.builder()
-                .email("admin@abc.com")
-                .isEnabled(true)
-                .isAccountNonExpired(true)
-                .isAccountNonLocked(true)
-                .isCredentialsNonExpired(true)
-                .password("admin")
-                .build()));
-    }
 
-    // this is just an alias for spring security
-    public Optional<User> findByUsername(String username) {
-        return findByEmail(username);
-    }
+        createUser(
+                new User(
+                        "sofiene",
+                        "thabet",
+                        "sof@gmail.com",
+                        "1234",
+                        UserRole.USER
+                )
+        );
+
+        createUser(
+                new User(
+                        "Ranim",
+                        "Naimi",
+                        "ranoum@gmail.com",
+                        "1234",
+                        UserRole.ADMIN
+                )
 
-    public Optional<User> findByEmail(String email) {
-        return userRepository.findByEmail(email);
+        );
     }
 
-	public void checkUserInput(User user) throws IllegalArgumentException {
-        throw new IllegalArgumentException("User input not valid");
+    @Override
+    public UserDetails loadUserByUsername(String email ) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() ->
+                        new UsernameNotFoundException(
+                                String.format(USER_NOT_FOUND_MSG, email)
+                        ));
     }
 
-    public User createUser(User user) {
-        user.setPassword(passwordEncoder.encode(user.getPassword()));
-        return userRepository.save(user);
+    public String createUser(User user) {
+        boolean userExists = userRepository.findByEmail(user.getEmail()).isPresent();
+
+        if (userExists) {
+            throw new IllegalStateException("email already exists."); // TODO: need better error handling
+        }
+
+        String encodedPassword = bCryptPasswordEncoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
+
+        userRepository.save(user);
+
+        // TODO: Send confirmation token
```

Comment:
This method is not supposed to manage tokens. It can only return confirmation about whether the user is created or not (boolean for e.g.). `RegistrationService`, however, can do that.

Reply:
Sorry I wrote that TODO by mistake.